File: lekcje/r_22-23/cpp/maks_roznica/main.cpp
#include <algorithm>
#include <iostream>
using namespace std;

int main() {
    int ilosc_liczb;
    cin >> ilosc_liczb;
    int ciag_liczb[ilosc_liczb];
    for (int i = 0; i < ilosc_liczb; i++) {
        cin >> ciag_liczb[i];
    }
    sort(ciag_liczb, ciag_liczb + ilosc_liczb);
    int roznica = ciag_liczb[ilosc_liczb - 1] - ciag_liczb[0];
    cout << roznica << endl;
}

File: lekcje/r_22-23/rust/maks_roznica/src/main.rs
use std::io;

fn maks_roznica(mut liczby: Vec<i32>, ilosc_liczb: usize) -> i32 {
    liczby.sort();
    let roznica = liczby[ilosc_liczb - 1] - liczby[0];
    roznica
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let ilosc_liczb: usize = input.trim().parse().unwrap();
    input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let mut liczby: Vec<i32> = Vec::new();
    for liczba in input.split_whitespace() {
        let liczba: i32 = liczba.trim().parse().unwrap();
        liczby.push(liczba);
    }
    let roznica = maks_roznica(liczby, ilosc_liczb);
    println!("Różnica {}", roznica)
}
